inc/elasticsearch_packages/namespace.php

```diff
@@ -0,0 +1,342 @@
+	}
+	if ( $existing_package_id ) {
+		// The cleanup does not have to happen straight away so postpone it for an hour
+		// to let the settings updates and data indexing complete.
+		wp_schedule_single_event( time() + HOUR_IN_SECONDS, 'altis.search.dissociate_package', [
+			$existing_package_id,
+		] );
+	}
+
+	try {
+		// Derive S3 bucket and path.
+		preg_match( '#^s3://([^/]+)/(.*?)$#', $file, $s3_file_parts );
+		$s3_bucket = $s3_file_parts[1];
+		$s3_key = $s3_file_parts[2];
+
+		// Create a new package.
+		$result = $client->createPackage( [
+			'PackageName' => $name, // required.
+			'PackageDescription' => $file,
+			'PackageSource' => [ // required.
+				'S3BucketName' => $s3_bucket,
+				'S3Key' => $s3_key,
+			],
+			'PackageType' => 'TXT-DICTIONARY',
+		] );
+
+		$package = $result['PackageDetails'];
+		$status = $package['PackageStatus'];
+
+		// Set package status.
+		if ( $for_network ) {
+			update_site_option( "altis_search_package_status_{$slug}", $status );
+			update_site_option( "altis_search_package_error_{$slug}", $package['ErrorDetails'] ?? null );
+		} else {
+			update_option( "altis_search_package_status_{$slug}", $status );
+			update_option( "altis_search_package_error_{$slug}", $package['ErrorDetails'] ?? null );
+		}
+
+		// Queue up package status check & association routine.
+		if ( in_array( $status, [ 'AVAILABLE', 'COPYING', 'VALIDATING' ], true ) ) {
+			wp_schedule_single_event( time() + 10, 'altis.search.check_package_status', [
+				$package_id,
+				$slug,
+				$for_network,
+			] );
+		}
+	} catch ( Exception $e ) {
+		// phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		trigger_error( 'Error creating package on Elasticsearch Service: ' . $e->getMessage(), E_USER_WARNING );
+		return null;
+	}
+
+	$package_id = sprintf( 'analyzers/%s', $package['PackageID'] );
+
+	// Return the AES package ID string.
+	return $package_id;
+}
+
+/**
+ * Check whether the package has finished associating with the AES domain
+ * before returning it.
+ *
+ * @param string|null $package_id The stored package ID.
+ * @param string $slug The slug for the package.
+ * @param boolean $for_network Whether this is a network level package.
+ * @return string|null
+ */
+function get_package_id( ?string $package_id, string $slug, bool $for_network = false ) {
+	// Check status.
+	if ( $for_network ) {
+		$status = get_site_option( "altis_search_package_status_{$slug}", 'COPYING' );
+	} else {
+		$status = get_option( "altis_search_package_status_{$slug}", 'COPYING' );
+	}
+
+	if ( $status !== 'ACTIVE' ) {
+		return null;
+	}
+
+	return $package_id;
+}
+
+/**
+ * Check and update the package association status.
+ *
+ * @param string $package_id The package ID to look up.
+ * @param string $slug The package slug.
+ * @param boolean $for_network Whether this is a network level package.
+ * @return void
+ * @throws Exception Thrown if there is any issue contacting the AES API, exceptions are caught.
```

Comment:
It doesn't look this like this throw?

Reply:
No I think this was added because PHPCS was being a bit dumb